stdlib/2and3/zipfile.pyi

```diff
@@ -0,0 +1,94 @@
+# Stubs for zipfile (Python 3.4)
+
+from typing import Callable, IO, Optional, Tuple, Type, Union
+from types import TracebackType
+import sys
+
+
+_SZI = Union[str, ZipInfo]
+_DT = Tuple[int, int, int, int, int, int]
+
+
+class BadZipFile(Exception): ...
+
+if sys.version_info > (3,):
+    BadZipfile = ...  # type: Type[BadZipFile]
+
+class LargeZipFile(Exception): ...
+
+class ZipFile:
+    debug = ...  # type: int
+    comment = ...  # type: bytes
+    def __init__(self, file: Union[str, IO[bytes]], mode: str = ..., compression: int = ...,
+                 allowZip64: bool = ...) -> None: ...
+    def __enter__(self) -> ZipFile: ...
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[Exception],
+                 exc_tb: Optional[TracebackType]) -> bool: ...
+    def close(self) -> None: ...
+    def getinfo(self, name: str) -> None: ...
+    def infolist(self) -> List[ZipInfo]: ...
+    def namelist(self) -> List[str]: ...
+    def open(self, name: _SZI, mode: str = ...,
+             pwd: Optional[bytes] = ...) -> IO[bytes]: ...
+    def extract(self, member: _SZI, path: Optional[_SZI] = ...,
+                pwd: bytes = ...) -> str: ...
+    def extractall(self, path: Optional[str] = ...,
+                   members: Optional[List[str]] = ...,
+                   pwd: Optional[bytes] = ...) -> None: ...
+    def printdir(self) -> None: ...
+    def setpassword(self, pwd: bytes) -> None: ...
+    def read(self, name: _SZI, pwd: Optional[bytes] = ...) -> bytes: ...
+    def testzip(self) -> Optional[str]: ...
+    def write(self, filename: str, arcname: Optional[str] = ...,
+              compress_type: Optional[int] = ...) -> None: ...
+    if sys.version_info >= (3,):
+        def writestr(self, zinfo_or_arcname: _SZI, data: Union[bytes, str],
+                     compress_type: Optional[int] = ...) -> None: ...
+    else:
+        def writestr(self,  # type: ignore
```

Comment:
Presumably because mypy still doesn't understand these version checks. We
will fix it, hopefully in release 0.4.4 (not in 0.4.3 which is about to go
out later this week).